Add customization to the information lines

Furkanzmc opened the pull request:
I added `g:infowindow_lines` and `g:infowindow_timeout` variables to customize the information that we display in the information window. If the timeout is not a positive number then the information window will not close automatically.

It's my first time contributing to a plugin, so I'm not *that* familiar with writing plugin rules. I'd appreciate any feedback.

Furkanzmc committed:
Add customization to the information lines

I added `g:infowindow_lines` and `g:infowindow_timeout` variables to
customize the information that we display in the information window.
If the timeout is not a positive number then the information window will
not close automatically.

Furkanzmc committed:
Reuse the same window for the consecutive calls

Furkanzmc committed:
Update the window configuration with consecutive calls

I forgot to include a call to nvim_win_set_config.

mcchrish commented:
@Furkanzmc great work!
I see you have removed the `InfoWindowToggle` command. Would like to still have that functionality to easily be able to map a single key to show and close the window.

mcchrish commented on autoload/infowindow.vim:
```diff
@@ -1,34 +1,64 @@
-  let numlines = ' lines:  ' . line('$') . ' '
+endfunction
+
+function infowindow#timer_handler(timer)
```
Some functions are probably not needed to be public if they are meant to be used internally. So instead of `infowindow#timer_handler` you can name it `s:time_handler` and call it like `<SID>time_handler`

Furkanzmc replied:
I could not convert `timer_handler` to a local function. `timer_start` does not allow it.

Furkanzmc commented:
> @Furkanzmc great work!
> I see you have removed the `InfoWindowToggle` command. Would like to still have that functionality to easily be able to map a single key to show and close the window.

I removed it because I added a timer to show and hide, but I'll add it back with no timeout,.

Furkanzmc committed:
Add back the toggle command

Furkanzmc committed:
Convert setup_window to local function

Furkanzmc commented:
Please check out the latest changes and let me know what you think.

mcchrish committed:
reset timer after manually closing window

mcchrish commented:
Thanks! I've adjusted a bit where the timer is still active even after manually closing the window.
